packages/hashtower/src/hashtower-hash-chain-proof-builder.ts

```diff
@@ -0,0 +1,57 @@
+import checkParameter from "./checkParameter"
+import { HashFunction, HashTowerHashChainProof } from "./types"
+import _add from "./add"
+import _indexOf from "./indexOf"
+import _build from "./build"
+
+/**
+ * HashTowerHashChainProofBuilder is a TypeScript implementation of HashTower to generate proofs of membership.
+ */
+export default class HashTowerHashChainProofBuilder {
+    private readonly _H: number
+    private readonly _W: number
+    private readonly _digestFunc: HashFunction
```

Comment:
I'd almost like this `HashFunction` type to be replaced with `(bigint, bigint) => bigint`. I really dislike custom types that are _super_ thin wrappers. It just adds more things I have to figure out.

`(bigint, bigint) => bigint` can be understood at a glance.

This is made 10x worse when the type definitions are in a different file.

Reply:
Replaced.